src/test/java/com/checkout/payments/PaymentsClientImplTest.java

```diff
@@ -79,6 +84,41 @@ void shouldRequestPayment() throws ExecutionException, InterruptedException {
 
     }
 
+    @Test
+    void shouldRequestPayment_customerSource() throws ExecutionException, InterruptedException {
+
+        CustomerSource customerSource = new CustomerSource(10, Currency.USD);
+
+        final PaymentRequest request = mock(PaymentRequest.class);
+        request.setSource(customerSource);
+
+        final PaymentResponse response = mock(PaymentResponse.class);
+
+        when(apiClient.postAsync(eq(PAYMENTS_PATH), any(SdkAuthorization.class), eq(PaymentResponse.class), eq(request), isNull()))
+                .thenReturn(CompletableFuture.completedFuture(response));
+
+        final CompletableFuture<PaymentResponse> future = paymentsClient.requestPayment(request);
+
+        assertNotNull(future.get());
+        assertEquals(response, future.get());
+
+    }
+
+    private class CustomerSource extends AbstractRequestSource
```

Comment:
I see what you are trying to do.
Hmmm, I don't have an opinion to adding inner classes for testing.

My suggestions are:
-In Java properties goes first, then constructors.